tables/endpointsecurity/tests/fileeventstableplugin.cpp

```diff
@@ -0,0 +1,116 @@
+#include "fileeventstableplugin.h"
+
+#include <catch2/catch.hpp>
+
+namespace zeek {
+namespace {
+IEndpointSecurityConsumer::Event
+generateEvent(IEndpointSecurityConsumer::Event::Type type) {
+  IEndpointSecurityConsumer::Event event;
+  event.header.timestamp = 1U;
+  event.header.parent_process_id = 2U;
+  event.header.orig_parent_process_id = 3U;
+  event.header.process_id = 4U;
+  event.header.user_id = 5U;
+  event.header.group_id = 6U;
+  event.header.platform_binary = true;
+  event.header.signing_id = "SigningID";
+  event.header.team_id = "TeamID";
+  event.header.cdhash = "12345";
+  event.header.path = "/path/to/application";
+  event.header.file_path = "/path/to/file";
+
+  if (type == IEndpointSecurityConsumer::Event::Type::Open) {
+    event.type = type;
+
+  } else if (type == IEndpointSecurityConsumer::Event::Type::Create) {
+    event.type = type;
+
+  } else {
+    throw std::logic_error("Invalid event type specified");
```

Comment:
Since we are in a test, why not the following?
```suggestion
    FAIL("Invalid event type specified");
```